packages/paste-core/utilities/grid/src/Grid.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {useTheme} from '@twilio-paste/theme';
+import {ThemeShape} from '@twilio-paste/theme-tokens';
+import {MarginProps, Space, SpaceProps} from '@twilio-paste/style-props';
+import {safelySpreadBoxProps} from '@twilio-paste/box';
+import {Flex, Vertical} from '@twilio-paste/flex';
+
+export interface GridProps extends SpaceProps {
+  children: NonNullable<React.ReactNode>;
+  gutter?: Space;
+  vertical?: Vertical;
+}
+
+const getNegativeMargin = ({gutter}: GridProps, theme: ThemeShape): {} => {
+  if (Array.isArray(gutter)) {
+    return (gutter as []).map((value: string) => {
+      return `-${theme.space[value]}`;
+    });
+  }
+
+  if (gutter) {
+    return `-${theme.space[gutter as string]}`;
+  }
+
+  return 'auto';
+};
+
+const getGridStyles = (props: GridProps): MarginProps => {
+  const theme = useTheme();
+  const marginStyles: MarginProps = {
+    marginLeft: getNegativeMargin(props, theme),
+    marginRight: getNegativeMargin(props, theme),
+  };
+
+  return marginStyles;
+};
+
+const Grid: React.FC<GridProps> = ({children, gutter, marginTop, marginBottom, vertical, ...props}) => {
+  const count = React.Children.count(children);
```

Comment:
You should move this inside the useMemo function, so we don't call it every re-render when GridColumns doesn't need to update.

A good rule of thumb is to just put everything related to hooks (useEffect, useMemo, useCallback, etc) inside the hook function.  It makes the code much more resilient because it's clear what the dependencies are for it to work simply due to code proximity.  I think it's recommended on the React website somewhere too.